src/sonic_ax_impl/mibs/ieee802_1ab.py

```diff
@@ -306,10 +306,16 @@ def reinit_data(self):
             logger.debug("Got mgmt ip from db : {}".format(self.mgmt_ip_str))
         try:
             addr_subtype_sub_oid = 4
-            mgmt_ip_sub_oid = (addr_subtype_sub_oid, *[int(i) for i in self.mgmt_ip_str.split('.')])
+            mgmt_ip_sub_oid = None
+            for mgmt_ip in self.mgmt_ip_str.split(','):
+                if '.' in mgmt_ip:
+                    mgmt_ip_sub_oid = (addr_subtype_sub_oid, *[int(i) for i in mgmt_ip.split('.')])
+                    break
         except ValueError:
             logger.error("Invalid local mgmt IP {}".format(self.mgmt_ip_str))
             return
+        if mgmt_ip_sub_oid == None:
+            return
         sub_oid = (ManAddrConst.man_addr_subtype_ipv4,
                    *mgmt_ip_sub_oid)
         self.man_addr_list.append(sub_oid)
```

Comment:
>man_addr_list [](start = 13, length = 13)

This list is keep growing.

Reply:
I guess you need to initialize it with empty list at the beginning of this function.

---
In reply to: [279029438](https://github.com/Azure/sonic-snmpagent/pull/106#discussion_r279029438) [](ancestors = 279029438)

Reply:
but line 291 do this initialization, do I need to change this self.man_addr_list to null list when return ?

Reply:
Let me explain this way. reinit_data() is supposed to run every 60 seconds, every time it will append one more sub_id, even the new sub_id is exactly same as before. So this process will eat many memory after days.

---
In reply to: [279134730](https://github.com/Azure/sonic-snmpagent/pull/106#discussion_r279134730) [](ancestors = 279134730)

Reply:
Thanks very much, I modified code to fix the issue, please review again.